src/netif.c

```diff
@@ -2000,7 +2017,8 @@ static inline int netif_deliver_mbuf(struct rte_mbuf *mbuf,
     assert(dev != NULL);
 
     pt = pkt_type_get(eth_type, dev);
-    if (NULL == pt) {
+
+    if (!(dev->flag & NETIF_PORT_FLAG_FORWARD2KNI) && NULL == pt) {
```

Comment:
if `err = pt->func(mbuf, dev);` returns `EDPVS_KNICONTINUE`, the mbuf will inject to `kni` again. pls also add the check `!(dev->flag & NETIF_PORT_FLAG_FORWARD2KNI)` to that place.
or the better way I think, to add a parameter `bool kni` to this API like
```
 static inline int netif_deliver_mbuf(struct rte_mbuf *mbuf,
                                      uint16_t eth_type,
                                      struct netif_port *dev,
                                      struct netif_queue_conf *qconf,
                                      bool kni)
```
Then if the mbuf is already copied to kni (in `lcore_job_recv_fwd`) , do not inject again.